beacon-chain/rpc/eth/beacon/validator.go

```diff
@@ -279,8 +280,8 @@ func valContainersByRequestIds(state state.BeaconState, validatorIds [][]byte) (
 				valIndex = primitives.ValidatorIndex(index)
 			}
 			validator, err := state.ValidatorAtIndex(valIndex)
-			if _, ok := err.(*statenative.ValidatorIndexOutOfRangeError); ok {
-				// Ignore well-formed yet unknown indexes.
+			if err != nil && strings.Contains(err.Error(), "out of bounds") {
```

Comment:
Is it possible to use [errors.Is](https://pkg.go.dev/errors#Is) here? 

You would have to change the [code that throws the error](https://github.com/prysmaticlabs/prysm/blob/79bbb9fe4ce84f200873d824ab913bb356e354a2/beacon-chain/state/state-native/getters_validator.go#L102) as well to wrap it.

```go
	if uint64(len(b.validators)) <= uint64(idx) {
		return nil, errors.Wrap(ErrOutOfBounds, fmt.Errorf("validator index %d does not exist", idx))
	}
```

then you could do this: 
```suggestion
			if errors.Is(err, statenative.ErrOutOfBounds) {
```

This pattern should be a bit more robust in future changes.

Reply:
+1. `strings.Contains` scares me due to brittle

Reply:
fixed